extensions/kind/src/create-cluster.ts

```diff
@@ -57,7 +63,19 @@ export async function setupIngressController(clusterName: string) {
   );
 }
 
-export async function connectionAuditor(provider: string): Promise<AuditResult> {
+export async function connectionAuditor(provider: string, items: AuditRequestItems): Promise<AuditResult> {
+  const image = items['kind.cluster.creation.controlPlaneImage'];
+  if (image && !image.includes('@sha256:')) {
+    return {
+      records: [
+        {
+          type: 'warning',
+          record: 'It is recommend to include the @sha256:{image digest} for the image used.',
+        } as AuditRecord,
+      ],
+    } as AuditResult;
+  }
+
```

Comment:
we should add all records together and not return the first one available

something like
```ts
 const records: AuditRecord[] = [];
  const auditResult = {
    records: records,
  };
  const image = items['kind.cluster.creation.controlPlaneImage'];
  if (image && !image.includes('@sha256:')) {
    records.push({
          type: 'warning',
          record: 'It is recommend to include the @sha256:{image digest} for the image used.',
        });
  }

  const providerSocket = extensionApi.provider
    .getContainerConnections()
    .find(connection => connection.connection.type === provider);

  if (!providerSocket) {
    return auditResult;
  }

  const memTotal = await getMemTotalInfo(providerSocket.connection.endpoint.socketPath);

  // check if configured memory is less than 6GB
  if (memTotal < 6000000000) {
      records.push({
          type: 'info',
          record: 'It is recommend to install Kind on a virtual machine with at least 6GB of memory.',
      });
  }
  return auditResult;
```